src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -74,6 +77,21 @@ class LogInWithShortLivedTokenPage extends Component {
             Log.info('[LoginWithShortLivedTokenPage] exitTo is workspace/new - handling new workspace creation in AuthScreens');
             return;
         }
+        this.navigateToExitRoute();
+    }
+
+    componentDidUpdate() {
+        this.navigateToExitRoute();
+    }
+
+    navigateToExitRoute() {
+        if (!this.props.betas) {
+            // Wait to navigate until the betas are loaded. Some pages like ReimbursementAccountPage require betas, so keep loading until they are available.
```

Comment:
**thought:** This would be easier to understand if we do not use `componentDidUpdate()` to mean "wait to navigate until the betas are loaded". It's not clear what specifically we are "waiting" for or where it happens. Who calls `getBetas()` and why are we waiting for that to happen here? 

**suggestion:** The comment about `ReimbursementAccountPage` creates a non-code dependency that should be changed as it's impossible to keep up to date. If `ReimbursementAccountPage` some day has no need for the betas will anyone remember to update this comment?

Reply:
You are right that this is very fragile and doesn't make much sense. 

What do we do if someone who isn't on the betas required for the ReimbursementAccountPage tries to navigate there? Currently we kick them back to the home page immediately, but that is problematic if they will have access to that page once the betas load. We could instead show a loading screen on the ReimbursementAccountPage until the betas have loaded. Is that a good solution?

Reply:
Gonna think about this for a bit.